keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -257,36 +242,54 @@ Metadata:
   finalizers:
   - cosi.io/finalizer [3]
 spec:
-  bucketClassName: [4]
-  bucketRef: [5]
+  provisioner: [4]
+  releasePolicy: [5]
+  accessModes: [6]
+  supportedProtocols: [7]
+  bucketClassName: [8]
+  bucketRef: [9]
     name:
     namespace:
-  secretRef: [6]
+    uuid:
+    resourceVersion:
+  secretRef: [10]
     name:
     namespace:
-  objectBucketName: [7]
+  bucketIdentifier: [11]
+  parameters: [12]
 status:
-  bucketAttributes: <map[string]string> [8]
-  phase: [9]
+  message: [14]
+  phase: [15]
   conditions:
 ```
 1. `name`: Generated in the pattern of `<BUCKET-CLASS-NAME>'-'<RANDOM-SUFFIX>`. 
 1. `labels`: COSI controller adds the label to its managed resources for easy CLI GET ops.  Value is the driver name returned by GetDriverInfo() rpc. Characters that do not adhere to [Kubernetes label conventions](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) will be converted to ‘-’.
 1. `finalizers`: COSI controller adds the finalizer to defer Bucket deletion until backend deletion ops succeed.
+1. `provisioner`: The provisioner field defined in the BucketClass.  Used by sidecars to filter BucketContents.
+1. `releasePolicy`: Prescribes outcome of a Delete events. **Note:** In Brownfield and Static cases, *Retain* is mandated.
+    - _Delete_:  the bucket and its contents are destroyed
+    - _Retain_:  the bucket and its data are preserved with only abstracting Kubernetes being destroyed
+1. `accessModes`: Declares the level of access given to credentials provisioned through this class.     If empty, drivers may set defaults.
+1. `supportedProtocols`:  An array of protocols the associated object store supports (e.g. swift, s3, gcs, etc.). *Only* serves a descriptive purpose and is not verified.
 1. `bucketClassName`: Name of the associated `BucketClass`.
 1. `bucketRef`: the name & namespace of the associated `Bucket`.
-1. `secretRef`: the name and namespace of the source secret in the `Bucket`'s namespace.  This `secret` may be generated by the driver or created by an admin.
-1. `objectBucketName`: the name of the actual bucket in the object store.
-1. `bucketAttributes`: stateful data relevant to the managing of the bucket but potentially inappropriate user knowledge (e.g. user's IAM role name).
+    - `name` : the Bucket’s name
```

Comment:
This is basically an objectRef, which is being deprecated. Is this still what we want here?

Reply:
Yes, definitely.  This structure, and it's go counterpart, are well defined.  ObjectRefs were deprecated because they were loose in what they could describe, which is what made them very hard to validate and document.